### PR TITLE
Format Azure image data according to schema

### PR DESCRIPTION
**Pull request by FKolwa**
Added:
- parse_image_version method to extract major.minor version and date from Azure image "version" property.
- format_image / format_images to convert raw image data to comply with JSON schema.

Additional changes:
- required "arch" property as part of the azure JSON schema definition.

Solves #94

### CODE REVIEW

File: src/rhelocator/update_images/azure.py [hunk 1]
```diff
@@ -229,3 +231,81 @@ def get_images() -> list[dict[str, str]]:
                     results.append(result)
 
     return results
+
+
+def parse_image_version(image_version: str) -> dict[str, str]:
+    """Parse an AWS image name and return extra data about the image.
+
+    Regex101: https://regex101.com/r/yHB9jJ/1
```
Comment (FKolwa) [inline src/rhelocator/update_images/azure.py:239]: Please give me a sanity check for this regex. I'm not 100% sure if all RHEL images follow this precise formula but it seems to be consistent within the images we currently retrieve.

Comment (major) [reply src/rhelocator/update_images/azure.py:241]: Haha, nicely done here. It looks like you're leaving off the optional last two digits after the year. Hopefully that's what you intended. 😉

**Review by major — APPROVED** [review]
This looks good! I'm glad to see the same function names being re-used here. 👏🏻

File: src/rhelocator/update_images/azure.py (hunk 1 above)
Comment (FKolwa) [reply src/rhelocator/update_images/azure.py:241]: Haha yeah to be honest I haven't found a real use for the optional digits, but I guess we can revisit this at any time.

**Pull request merged**